### PR TITLE
copy deployment code over

### PR DESCRIPTION
**Pull request by ShivamChamoli**
--> Creation of the base operator code + deployment and service code

### CODE REVIEW

File: deploy/cr.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: "hive.example.com/v1alpha1"
+kind: "Hive"
+metadata:
+  name: "example"
+spec:
+  size: 3
```
Comment (csrwng) [inline deploy/cr.yaml]: what is the purpose of the size field?

File: kubeconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+apiVersion: v1
```
Comment (csrwng) [inline kubeconfig.yaml]: is this file needed here?

File: config/config.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+apiVersion: hive.example.com/v1alpha1
```
[the remark this answers is not in the document]
Comment (csrwng) [reply config/config.yaml]: Should the apiVersion of the operator CRD be different than the thing that it installs?

File: pkg/stub/handler.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
+package stub
```
[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply pkg/stub/handler.go]: This is part of the boiler plate code generated by the operator-sdk. I wouldn't change it, at least right now, when we all are new. Moreover, its used in the main.go to call the NewHandler function, so I personally think it should be a package called stub. Let me know if you feel differently, I'm sure It will not be a big change.

File: deploy/crd.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+kind: CustomResourceDefinition
+metadata:
+  name: hives.hive.example.com
+spec:
+  group: hive.example.com
+  names:
+    kind: Hive
+    listKind: HiveList
+    plural: hives
+    singular: hive
```
[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply deploy/crd.yaml]: Actually this just creates the CRD of kind Hive (with plural and singular configuration used to differentiate between multiple instances of the kind).

The configuration for the instance is listed in the cr.yaml as
apiVersion: "hive.example.com/v1alpha1"
kind: "Hive"
metadata:
  name: "example"
spec:
  size: 3
size being the replication.

File: deploy/cr.yaml (hunk 1 above)
Comment (ShivamChamoli) [reply deploy/cr.yaml]: Size is replication for our deployment. Check pkg/apis/hive/v1alpha1/types.go, it will have:-
type HiveSpec struct {
	Image string `json:"image"`
	Size  int32  `json:"size"`
}

File: kubeconfig.yaml (hunk 2 above)
Comment (ShivamChamoli) [reply kubeconfig.yaml]: Nope. removed it.

**Comment by dgoodwin** [comment]
Closing as we're off to https://github.com/openshift/hive-operator/pull/2 due to some git issues.

File: pkg/stub/handler.go [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+						Name:  "hive-operator",
+					}},
+				},
+			},
+		},
+	}
+}
+
+//Service
+func newHiveService(cr *v1alpha1.Hive) *corev1.Service {
```
[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply pkg/stub/handler.go]: Removed service code

File: README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+# Opertor for OpenShift Hive
```
[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply README.md]: Done.

File: config/config.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply config/config.yaml]: Done.

[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply config/config.yaml]: > Could you switch all apiVersions to "hive.openshift.io/v1alpha1".

done.

[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply config/config.yaml]: > Should the apiVersion of the operator CRD be different than the thing that it installs?

I couldn't catch that properly but if you mean why apiVersion field has different entries in crd.yaml and cr.yaml, then - basically crd.yaml when created will create a CRD for kind Hive and to create a CRD we need the version of the kubernetes API that provides it, in this case 'apiextensions.k8s.io/v1beta1'. Now cr.yaml is basically an instance of the above created kind (Hive) and hence it has the apiVersion as 'hive.openshift.io'. Do let me know if I am wrong.

[the remark this answers is not in the document]
Comment (csrwng) [reply config/config.yaml]: No, what I meant, is should the api version for the operator config be different than the api version of the hive resources. However, as I mentioned the other day, it is the same in other operators, so it's fine as it is.

[the remark this answers is not in the document]
Comment (ShivamChamoli) [reply config/config.yaml]: yup great :)
